src/figuresClasses.ts

```diff
@@ -2,10 +1,83 @@
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+type Colors = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+function checkError(condition: boolean, message: string): void {
+  if (condition) {
+    throw new Error(message);
+  }
+}
+
+export class Triangle implements Figure {
+  readonly shape = 'triangle';
+
+  constructor(
+    public color: Colors,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    checkError(a <= 0 || b <= 0 || c <= 0, 'One or more length <= 0');
+
+    checkError(
+      a + b <= c || a + c <= b || b + c <= a,
+      "Your sides can't form a triangle",
+    );
+  }
+
+  getArea(): number {
```

Comment:
According to the checklist, the 'getArea' function should not accept any parameters since all necessary values are already available as object fields.